source/CommonJobs/CommonJobs.Application.EvalForm/Commands/UpdateCalificationsCommand.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            _updateEvaluation = updateEvaluation;
+
+            _loggedUser = loggedUser;
+        }
+
+        public override void Execute()
+        {
+            bool updateEvaluationComments = false;
+            bool updateEvaluationProject = false;
+
+            var storedEvaluation = RavenSession.Load<EmployeeEvaluation>(_updateEvaluation.EvaluationId);
+            if (storedEvaluation == null)
+            {
+                throw new ApplicationException(string.Format("Error: Evaluación {0} inexistente", _updateEvaluation.EvaluationId));
+            }
+
+            foreach (var calification in _updateEvaluation.Califications)
+            {
+                var storedCalification = RavenSession.Load<EvaluationCalification>(calification.CalificationId);
+                if (storedCalification == null)
+                {
+                    throw new ApplicationException(string.Format("Error: Calificación {0} inexistente para evaluación {1}", calification.CalificationId, _updateEvaluation.EvaluationId));
+                }
+
+                // Checks if the user trying to update the califications can actually do it
+                if (CanUpdate(_loggedUser, storedEvaluation, storedCalification))
+                {
+                    UpdateCalification(calification, storedCalification, _updateEvaluation.Finished);
+
+                    // If it's the responsible's or the company's calification, then the evaluation project should be updated [ TODO: Improvement - Check if the project has changed before updating it ]
+                    updateEvaluationProject = (storedCalification.Owner == CalificationType.Responsible || storedCalification.Owner == CalificationType.Company);
+                    // If it's the company's calification (or devolution), then the evaluation comments should be updated [ TODO: Improvement - Check if the comments have changed before updating it ]
+                    updateEvaluationComments = storedCalification.Owner == CalificationType.Company;
+
+                    // If it's the responsible's calification and it's finished, then the company calification should be created
+                    CreateCompanyEvaluation(storedCalification);
+                }
+            }
+
+            // The evaluation gets updated only if the calification being updated is the company or the final one (devolution)
+            UpdateEvaluation(updateEvaluationComments, updateEvaluationProject, storedEvaluation);
+        }
+
+        private void UpdateCalification(UpdateCalificationDTO calification, EvaluationCalification storedCalification, bool finished)
+        {
+            // Update the calification comments and values
+            storedCalification.Comments = calification.Comments;
+            storedCalification.Califications = calification.Items;
+            storedCalification.Finished = finished;
```

Comment:
It could be possible to deal with `finish` field in a different way? At least to add a TODO mark to review it, I think that should be special commands or something to enforce the programmer to follow the flow.